tests/ui/self/elision/ref-self-multi.rs

```diff
@@ -0,0 +1,29 @@
+// run-pass
+
+#![feature(arbitrary_self_types)]
+#![allow(non_snake_case)]
+#![allow(unused)]
+
+use std::marker::PhantomData;
+use std::ops::Deref;
+
+struct Struct { }
+
+struct Wrap<T, P>(T, PhantomData<P>);
+
+impl<T, P> Deref for Wrap<T, P> {
+    type Target = T;
+    fn deref(&self) -> &T { &self.0 }
+}
+
+impl Struct {
+    fn ref_box_ref_Self(self: &Box<&Self>, f: &u32) -> &u32 {
```

Comment:
is that desirable? 🤔 would expect this to be ambiguous

Reply:
Yeah - these tests are confusing. This test passes _because_ lifetime of the self type is ambiguous, so the elision rules instead assume we want to return the lifetime of the `f`, which is what we do.

We could instead say "lifetime of the self type is unclear, elision is impossible, manually specify a lifetime" - I _think_ I tried that and it broke a larger number of tests.

Reply:
alright :thinking: I think that our UI tests don't perfectly represent the way users use Rust, as `self: Something` feels like it should be overrepresented in our test suite.

My feelings here is: can we get a full and self-contained explanation of the elision rules rn, i.e. before this bug, and then the different alternatives :sweat_smile: I would have to dive into this issue myself to really have an opinion. But lifetime elision feels like something where we can work towards the "perfect" solution thanks to editions.